src/ucp/wireup/wireup_cm.c

```diff
@@ -1413,13 +1415,21 @@ ucp_ep_cm_connect_server_lane(ucp_ep_h ep, uct_listener_h uct_listener,
     uct_ep_params_t uct_ep_params;
     uct_ep_h uct_ep;
     ucs_status_t status;
+    unsigned max_num_paths;
 
     ucs_assert(ucp_ep_get_lane(ep, lane) == NULL);
 
     ucp_wireup_get_dst_rsc_indices(ep, &ucp_ep_config(ep)->key,
                                    remote_address, addr_indices,
                                    dst_rsc_indices);
 
+    /* Address should not be empty and contain transport addresses of a single
+     * device. Therefore dev_num_paths value is expected to be the same for all
+     * address entries.
+     */
+    ucs_assert(remote_address->address_count > 0);
+    max_num_paths = remote_address->address_list[0].dev_num_paths;
```

Comment:
1. Imo, this will unnecessray coplicate the code, because we expect that there will be always a single device transports
2. Can do that, but it would look less intuitive

Reply:
1. can we check/assert it?